site-builder/src/site/resource.rs

```diff
@@ -344,15 +344,17 @@ impl ResourceManager {
             extension.to_string_lossy(),
             full_path.to_string_lossy()
         ))?)?;
-        // TODO(giac): this could be (i) async; (ii) pre configured with the number of shards to
-        //     avoid chain interaction (maybe after adding `info` to the JSON commands).
-        let output = self.walrus.blob_id(full_path.to_owned(), None)?;
-        // Need to check the size to avoid calling encode on empty files.
+
         let plain_content = std::fs::read(full_path)?;
         if plain_content.is_empty() {
-            // We are ignoring empty files.
-            return Ok(None);
+            let error_message = format!("The file {} is empty.", full_path.to_string_lossy());
+            return Err(
+                anyhow!(error_message).context("Walrus sites does not support empty files.")
```

Comment:
Nit: technically, it is Walrus that does not support empty files.

```suggestion
                anyhow!(error_message).context("Walrus does not support empty files.")
```